### PR TITLE
Fix #1554: Improve compare_config for NXOS partial merging

### LINKED ISSUE #1554: Partial Idempotent Configuration Push
### Description of Issue/Question

Add the ability to push partial configurations idempotently. 

Napalm provides two greats configuration deployment abilities:
* Config Replace
* Config Merge

While it is clear, Config Replace is more powerful, it is also harder to get to the maturity of full configuration deployment. Often times in customer implementations, Config Merge is being used. 

In Netutils, there is the capability to understand what configurations are missing from a potential Merge. You can see an example of this in the tests there: https://github.com/networktocode/netutils/tree/develop/tests/unit/mock/config/compliance/diff_network_config/cisco_ios

My proposal is to add another method to idempotently push partial configurations. This is actually the norm in Ansible, and should provide better integrations to get to the preferred Config Replace strategy. 

I am still thinking of appropriate naming of the method, but hopefully that provides some context. 

### Did you follow the steps from https://github.com/napalm-automation/napalm#faq
(Place an ``x`` between the square brackets where applicable)

- [x] Yes
- [] No

### napalm version
N/A (New Feature)

### Network operating system version
(Paste verbatim output from `show version` - or equivalent - between quotes below)

N/A (New Feature)

### Steps to Reproduce the Issue
N/A (New Feature)

### Error Traceback
(Paste the complete traceback of the exception between quotes below)

N/A (New Feature)

### PR DESCRIPTION
**Pull request by cmsirbu**
NXOS partial merges were not aware of command hierarchy when showing the differences to be applied via the `compare_config` method. We (myself and @mundruid) have changed the merge diffing to use the [netutils](https://netutils.readthedocs.io/en/latest/) package. Docstrings and documentation updated as well.

Tests would be a stretch goal for another day since this code path is not tested at the moment.

See an example below of old and new behavior:

```
Intended configuration:

candidate_cfg = """
interface loopback0
  ip address 10.1.4.5/32
  ip router ospf 100 area 0.0.0.1
"""

Base - full configuration on device with the relevant section below:
...
interface loopback0
  ip address 10.1.4.4/32
  ip router ospf 100 area 0.0.0.1
...

nxos1.load_merge_candidate(config=candidate_cfg)

Before:

print(nxos1.compare_config())
  ip address 10.1.4.5/32


With netutils, respecting required command hierarchy:

print(nxos1.compare_config())
interface loopback0
  ip address 10.1.4.5/32
```

Fixes #1554

### CODE REVIEW

File: napalm/nxos/nxos.py [hunk 1]
```diff
@@ -209,22 +210,29 @@ def _commit_merge(self) -> None:
...
         """
-        diff = []
         running_config = self.get_config(retrieve="running")["running"]
-        running_lines = running_config.splitlines()
-        for line in self.merge_candidate.splitlines():
-            if line not in running_lines and line:
-                if line[0].strip() != "!":
-                    diff.append(line)
-        return "\n".join(diff)
+        return diff_network_config(self.merge_candidate, running_config, "cisco_nxos")
```
Comment (mirceaulinic) [inline napalm/nxos/nxos.py:235]: I'm not entirely familiar with what's this function doing, but looking at https://netutils.readthedocs.io/en/latest/netutils/configs/index.html?highlight=diff_network_config#netutils.config.compliance.diff_network_config sounds great!

I think the linter is failing on this line (see https://github.com/napalm-automation/napalm/runs/5176237179?check_suite_focus=true) then we should be gtg! 😄

Comment (mundruid) [reply napalm/nxos/nxos.py]: Fixed. Should have ran autopep8. 😄

Comment (mirceaulinic) [reply napalm/nxos/nxos.py]: Also `black .` please: https://github.com/napalm-automation/napalm/runs/5176322883?check_suite_focus=true 😄

Comment (cmsirbu) [reply napalm/nxos/nxos.py]: I think autopep8 solved more than it should have, breaking black. :)

File: docs/support/nxos.rst [hunk 2]
```diff
@@ -66,4 +66,30 @@ As a result, merges are **not atomic**.
...
 _____
 
-Diffs for merges are simply the lines in the merge candidate config.
+Diffs for merges are simply the lines in the merge candidate config. 
+[Netutils](https://netutils.readthedocs.io/en/latest/) is used for creating the merge diff between the candidate and running configurations. 
+One caveat of using netutils diff of configurations is that the diff is performed offline and not online in the device.
+
+Example assuming that the device config contains:
+
+```
```
Comment (mirceaulinic) [inline docs/support/nxos.rst:75]: One more: the doc build is failing on this, it seems: please indent and optionally add here `.. code-block:: text` (for the RST format). Same below, likely: `.. code-block:: python`.

Comment (cmsirbu) [reply docs/support/nxos.rst]: Yeah, it was markdown mixed in, I converted it to reST now.

**Review by mirceaulinic — APPROVED** [review]

**Pull request merged**